Airbrakes/airbrakes.py

```diff
@@ -38,19 +38,29 @@ def __init__(self, logger: Logger, servo: Servo, imu: IMU):
 
     def update(self):
         """
-        Called every loop iteration. Depending on the current state, it will do different things. It
-        is what controls the airbrakes and chooses when to move to the next state.
+        Called every loop iteration from the main process. Depending on the current state, it will
+        do different things. It is what controls the airbrakes and chooses when to move to the next
+        state.
         """
         # Gets the current extension and IMU data, the states will use these values
         self.current_extension = self.servo.current_extension
-        self.current_imu_data = self.imu.get_imu_data()
+        data_packets = []
+
+        # Let's get 50 data packets to ensure we have enough data to work with.
+        # 50 is an arbitrary number for now - if the time resolution between each data packet is
+        # 2ms, then we have 2*50 = 100ms of data to work with at once.
+        while len(data_packets) < 50:
+            # get_imu_data() gets the "first" item in the queue, i.e, it *may* not be the most
```

Comment:
Queues are FIFO so it will really be the oldest data that you get first, but its fine bc we have a time stamp

Reply:
yeah, we _could_ process on the latest data, but I fear potentially having discarded data packets which might be critical to determining the state, and just logging in general.